### PR TITLE
holo registration server profile

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

File: overlays/holo-nixpkgs/holo-registration-service/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+```
+MONGO_URI=mongodb://127.0.0.1:27017/?ssl=false
+```
+
+You need to be running a mongoDB locally if you are not already hosting a remote db.
+
+## Requirements
+
+- Node
+- MongoDB
```
Comment (peeech) [inline overlays/holo-nixpkgs/holo-registration-service/README.md:57]: Dev only

File: overlays/holo-nixpkgs/holo-registration-service/src/const.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const UNIX_SOCKET = process.env.NODE_ENV === 'test' ? 8800 : '/run/holo-registration-service/holo-registration-service.sock'
+
+const HAPP_PORT = 42233
+
+const DEV_UID_OVERRIDE = process.env.DEV_UID_OVERRIDE
+
+const MONGO_URI = process.env.MONGO_PATH ? fs.readFileSync(process.env.MONGO_PATH, 'utf8') : 'mongodb://127.0.0.1:27017/?ssl=false'
+
+const CONFIGURE_HC = process.env.NODE_ENV === 'test' ? './tests/config.yaml' : '/var/lib/configure-holochain/config.yaml'
```
Comment (peeech) [inline overlays/holo-nixpkgs/holo-registration-service/src/const.js:12]: Naybe read it from env var set by Nixos profile? For consistency with other profiles created in holo-nixpkgs.

File: profiles/logical/holo/holo-registration-server/default.nix [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+          proxyWebsockets = true;
+        };
+      };
+      serverName = "holo-registration-service.holo.host";
+    };
+
+    virtualHosts.localhost = {
+        locations."/holochain-api/".proxyPass = "http://unix:/run/hpos-holochain-api/hpos-holochain-api.sock:/";
+        locations."/holo-registration-service/".proxyPass = "http://${registrationServiceApiSocket}:/";
+      };
```
Comment (peeech) [inline profiles/logical/holo/holo-registration-server/default.nix:59]: indentation

File: profiles/logical/holo/holo-registration-server/default.nix [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  users.groups.apis = {};
+
+  users.users.nginx.extraGroups = [ "apis" ];
+
+  services.hpos-holochain-api.enable = true; # Temporary
+
+  services.ssh-pinger = {
+    enable = true;
+    credentialsDir = registrationServerCredentialsDir;
+  };
```
Comment (peeech) [inline profiles/logical/holo/holo-registration-server/default.nix:129]:
```suggestion
```

**Review by peeech — CHANGES_REQUESTED** [review]
A few small requests for change

**Review by peeech — APPROVED** [review]
Great!

File: overlays/holo-nixpkgs/holo-registration-service/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# holo-registration-service
+
+Node API service that verifies if a holo user is registered by Holo.ltd.
+// REMINDER: Tell zo-el should add more details
```
Comment (JettTech) [inline overlays/holo-nixpkgs/holo-registration-service/README.md:4]: > just pointing this out. Do we need me detail here?

File: overlays/holo-nixpkgs/holo-registration-service/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+## Methods:
+
+- `/register-user/` (POST): Validated the registration code and gets back the appropriate joining codes back based on the registration code and role provided.
```
Comment (JettTech) [inline overlays/holo-nixpkgs/holo-registration-service/README.md:8]:
```suggestion
- `/register-user/` (POST): Validated the registration code and gets back the appropriate joining codes based on the registration code and role provided.
```

File: overlays/holo-nixpkgs/holo-registration-service/README.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+- `Invalid Registration code status`
+  - Description: The registration code was possibly not sent to the user hence invalid
+- `Registration code already used`
+- `Registration code deleted`
+- `Could not generate mem-proof`
+- `Invalid registration code`
+  - Description: registration code not found in the db
+- `Invalid arguments passed`
+- `Something went wrong`
+  - Description: Error returned when something unexpected happens, see the `info` to lear more about this error
```
Comment (JettTech) [inline overlays/holo-nixpkgs/holo-registration-service/README.md:39]:
```suggestion
  - Description: Error returned when something unexpected happens, see the `info` to learn more about this error
```

File: profiles/logical/holo/holo-registration-server/default.nix [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    install-list = {
+      self_hosted_happs = [];
+      core_happs = [
+        {
+          app_id = "joining-code-factory";
+          bundle_url = "https://github.com/zo-el/temp/releases/download/v0.0.8/joining-code-factory.0_1_0_alpha1.happ";
+        }
+      ];
+    };
+    membrane-proofs = {
```
Comment (JettTech) [inline profiles/logical/holo/holo-registration-server/default.nix:117]: Can we add a temp flag for this too?

**Review by JettTech — APPROVED** [review]
Awesome work! Just found a couple typos, and asked for one small comment. 

:shipit:

File: overlays/holo-nixpkgs/holo-registration-service/README.md (hunk 5 above)
Comment (zo-el) [reply overlays/holo-nixpkgs/holo-registration-service/README.md]: Thanks lisa, I had made the updates just needed to remove this reminder.

**Pull request merged**